src/lib.rs

```diff
@@ -797,13 +797,13 @@ impl Url {
     pub fn with_default_port<F>(&self, f: F) -> io::Result<HostAndPort<&str>>
     where F: FnOnce(&Url) -> Result<u16, ()> {
         Ok(HostAndPort {
-            host: try!(self.host()
+            host: self.host()
                            .ok_or(())
-                           .or_else(|()| io_error("URL has no host"))),
-            port: try!(self.port_or_known_default()
+                           .or_else(|()| io_error("URL has no host"))?,
+            port: self.port_or_known_default()
                            .ok_or(())
                            .or_else(|()| f(self))
-                           .or_else(|()| io_error("URL has no port number")))
+                           .or_else(|()| io_error("URL has no port number"))?
```

Comment:
rustfmt does a lot more things than just replacing `try!`. If we’re gonna go down that road I’d prefer to go all the way and enforce rustfmt-like style in CI. But that seems more disruptive than I wanted to get into right now, so I recommended untry instead.